packages/godwoken/native/src/lib.rs

```diff
@@ -0,0 +1,115 @@
+use ckb_types::packed::Transaction;
+use gw_chain::{
+    chain::{
+        Chain, HeaderInfo, L1Action, L1ActionContext, ProduceBlockParam, SyncEvent, SyncParam,
+        TransactionInfo,
+    },
+    genesis,
+    next_block_context::NextBlockContext,
+    tx_pool::TxPool,
+};
+use gw_common::smt::{H256, SMT};
+use gw_common::sparse_merkle_tree::SparseMerkleTree;
+use gw_config::{Config, GenesisConfig};
+use gw_generator::{
+    backend_manage::BackendManage,
+    generator::{DepositionRequest, WithdrawalRequest},
+    Generator,
+};
+use gw_store::Store;
+use gw_types::{
+    bytes::Bytes,
+    packed::{L2Block, L2Transaction, RawL2Block, Script},
+    prelude::*,
+};
+use neon::prelude::*;
+use parking_lot::Mutex;
+use std::collections::HashMap;
+use std::fs::File;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::{Arc, RwLock};
+
+pub struct NativeChain {
+    pub config: Config,
+    // Deprecated
+    pub running: Arc<AtomicBool>,
+    pub chain: Arc<RwLock<Chain>>,
+}
+
+impl NativeChain {
+    pub fn running(&self) -> bool {
+        self.running.load(Ordering::Acquire)
+    }
+
+    pub fn stop(&self) {
+        self.running.store(false, Ordering::Release);
+    }
+}
+
+declare_types! {
+    pub class JsNativeChain for NativeChain {
+        init(mut cx) {
+            let config_path = cx.argument::<JsString>(0)?.value();
```

Comment:
I think genesis configuration can also be passed from JS side to Rust side. This way we can manage all configurations in JS, for a more consistent design.

Reply:
Have updated the TS definition to support configuration.